src/Authentication/Services/SystemRegisterService.cs

```diff
@@ -105,7 +105,7 @@ public Task<bool> SetDeleteRegisteredSystemById(string id, Guid systemInternalId
         /// <inheritdoc/>
         public Task<bool> UpdateWholeRegisteredSystem(RegisterSystemRequest updateSystem, string systemId, CancellationToken cancellationToken)
         {
-            return _systemRegisterRepository.UpdateRegisteredSystem(updateSystem);
+            return _systemRegisterRepository.UpdateRegisteredSystem(updateSystem, systemId, cancellationToken);
```

Comment:
Is there a reason why systemid parameter is sent here? I see you already validated the systemid on url and systmid on the updatesystem json.